src/ResourceManager/AzureBatch/Commands.Batch/Models/BatchClient.Files.cs

```diff
@@ -219,12 +222,35 @@ public void DownloadNodeFile(DownloadNodeFileOptions options)
                     }
             }
 
-            DownloadNodeFileByInstance(nodeFile, options.DestinationPath, options.Stream, options.AdditionalBehaviors);
+            DownloadNodeFileByInstance(nodeFile, options.DestinationPath, options.Stream, options.RangeStart, options.RangeEnd, options.AdditionalBehaviors);
         }
 
         // Downloads the file represented by an NodeFile instance to the specified path.
-        private void DownloadNodeFileByInstance(NodeFile file, string destinationPath, Stream stream, IEnumerable<BatchClientBehavior> additionalBehaviors = null)
+        private void DownloadNodeFileByInstance(NodeFile file, string destinationPath, Stream stream, long? rangeStart, long? rangeEnd, IEnumerable<BatchClientBehavior> additionalBehaviors = null)
         {
+            Protocol.RequestInterceptor interceptor = new Protocol.RequestInterceptor(baseRequest =>
+            {
+                var fromTaskRequest = baseRequest as BatchRequests.FileGetFromTaskBatchRequest;
+                if (fromTaskRequest != null && rangeStart != null && rangeEnd != null)
```

Comment:
So if they specify a start but not an end, or vice versa... does this behave correctly?  Finding it a bit hard to track if they get defaulted somewhere along the line, but since they are still nullable at this point I am kind of guessing not...?